chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/CertifiedProductDownloadableResourceCreatorJob.java

```diff
@@ -63,53 +62,54 @@ public void execute(final JobExecutionContext jobContext) throws JobExecutionExc
                 edition);
         try {
             List<CertifiedProductDetailsDTO> listings = getRelevantListings();
-
-            initializeWritingToFiles();
-
-            List<CompletableFuture<CertifiedProductSearchDetails>> futures =
-                    new ArrayList<CompletableFuture<CertifiedProductSearchDetails>>();
-
-            for (CertifiedProductDetailsDTO dto : listings) {
-                CompletableFuture<CertifiedProductSearchDetails> cpCompletableFuture =
-                    CompletableFuture.supplyAsync(new Supplier<CertifiedProductSearchDetails>() {
-                        @Override
-                        public CertifiedProductSearchDetails get() {
-                            CertifiedProductSearchDetails cpDTO = null;
-                            try {
-                                cpDTO = cpdManager.getCertifiedProductDetails(dto.getId());
-                                LOGGER.info("Finishing Details for: " + dto.getId());
-                            } catch (Exception e) {
-                                LOGGER.error(e);
+            
+            try (CertifiedProductXmlPresenter xmlPresenter = new CertifiedProductXmlPresenter();
+                    CertifiedProductCsvPresenter csvPresenter = getCsvPresenter()) {
+
+                initializeWritingToFiles(xmlPresenter, csvPresenter);
+
+                List<CompletableFuture<CertifiedProductSearchDetails>> futures =
+                        new ArrayList<CompletableFuture<CertifiedProductSearchDetails>>();
+    
+                for (CertifiedProductDetailsDTO dto : listings) {
+                    CompletableFuture<CertifiedProductSearchDetails> cpCompletableFuture =
+                        CompletableFuture.supplyAsync(new Supplier<CertifiedProductSearchDetails>() {
```

Comment:
Can this  new Supplier definition be moved out to be a separate class? I did something like that with some Comparators in the chpl-api project to make them static  at findbug's suggestion; not sure if that's possible here

Reply:
Change made